### PR TITLE
Live preview changes on Manage email page

### LINKED ISSUE #53696: Live preview changes on Manage email page
1. Implement live preview when managing a specific email for Subject, Email heading, and Additional content.

### PR DESCRIPTION
**Pull request by NeosinneR**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Part of #52228. Adds live preview when managing a specific email.

Closes #53696.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable **Email improvements** feature in **WooCommerce > Settings > Advanced > Features**.
2. Go to **WooCommerce > Settings > Email**.
3. Click **Manage** on any email.
4. Update the **Subject** field and check that the subject is updated in the email preview.
5. Update the **Email heading** field and check that the heading is updated in the email preview.
6. Update the **Additional content** field and check that it's updated in the email preview.

### CODE REVIEW

**Comment by alex-mpoet** [comment]
HTML tags disappear in the additional content's live preview after making any changes to it. Check the `wc_email_failed_order` email.

**Comment by NeosinneR** [comment]
@alex-mpoet ah, good catch! The same issue as in the email footer previously. Fixed in 45c89f1.

File: plugins/woocommerce/tests/php/src/Internal/Admin/EmailPreview/EmailPreviewTest.php [hunk 1]
```diff
@@ -238,4 +239,45 @@ public function test_transient_values_in_preview() {
...
+	 * Test that transient values are applied in subject
+	 */
+	public function test_transient_values_in_subject() {
+		$email_id = EmailPreview::DEFAULT_EMAIL_ID;
+		$key      = "woocommerce_${email_id}_subject";
+
+		$this->sut->set_email_type( EmailPreview::DEFAULT_EMAIL_TYPE );
+		$this->assertEquals( $this->sut->get_subject(), 'Your ' . self::SITE_TITLE . ' order has been received!' );
+
+		set_transient( $key, 'transient_subject', HOUR_IN_SECONDS );
```
Comment (alex-mpoet) [inline plugins/woocommerce/tests/php/src/Internal/Admin/EmailPreview/EmailPreviewTest.php:254]: Should we clear transient values after test? This one only affects the `EmailPreview` class, but still might be a good thing to do.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/settings-email.spec.js [hunk 2]
```diff
@@ -263,6 +263,33 @@ test.describe( 'WooCommerce Email Settings', () => {
 			await expect( page.getByLabel( 'Email preview type' ) ).toHaveCount(
 				0
 			);
+
+			// Change subject and observed it's changed in the preview
```
Comment (alex-mpoet) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/settings-email.spec.js]: Typo: observed => observe.

File: plugins/woocommerce/tests/php/src/Internal/Admin/EmailPreview/EmailPreviewTest.php (hunk 1 above)
Comment (NeosinneR) [reply plugins/woocommerce/tests/php/src/Internal/Admin/EmailPreview/EmailPreviewTest.php:254]: Fixed in 329a0af.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/settings-email.spec.js (hunk 2 above)
Comment (NeosinneR) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/settings-email.spec.js]: Fixed in 755d2b6.

**Review by alex-mpoet — APPROVED** [review]
Good job! And thank you for the prompt fixes!

**Comment by senadir** [comment]
Hey @NeosinneR this PR is causing a warning on PHP 8.3

```
Deprecated: Using ${var} in strings is deprecated, use {$var} instead in /Users/nadir/work/monorepo/plugins/woocommerce/includes/emails/class-wc-email.php on line 1239
```

File: plugins/woocommerce/includes/emails/class-wc-email.php [hunk 3]
```diff
@@ -1199,4 +1217,31 @@ private function clear_alt_body_field(): void {
...
+		/**
+		 * This filter is documented in templates/emails/email-styles.php
+		 *
+		 * @since 9.6.0
+		 * @param bool $is_email_preview Whether the email is being previewed.
+		 */
+		$is_email_preview = apply_filters( 'woocommerce_is_email_preview', false );
+		if ( $is_email_preview ) {
+			$email_id  = $this->id;
+			$transient = get_transient( "woocommerce_${email_id}_${key}" );
```
Comment (senadir) [inline plugins/woocommerce/includes/emails/class-wc-email.php:1239]: This should be 
```php
			$transient = get_transient( "woocommerce_{$email_id}_{$key}" );
```

Comment (NeosinneR) [reply plugins/woocommerce/includes/emails/class-wc-email.php:1239]: Thanks for the report, @senadir. I [created a fix](https://github.com/woocommerce/woocommerce/pull/53787) - would you like to review it, or should I ask someone else?

**Pull request merged**